python/tvm/tir/schedule/schedule.py

```diff
@@ -2111,6 +2112,82 @@ def after_unannotate(a: T.handle, b: T.handle) -> None:
             self, block_or_loop, ann_key
         )
 
+    ########## Schedule: Layout transformation ##########
+
+    @type_checked
+    def transform_layout(
+        self,
+        block: BlockRV,
+        buffer_index: int,
+        is_write_index: bool,
```

Comment:
If there is an advantage to keeping the three parameters to identify the buffer, can `is_write_index` be a `enum.Enum`?  From a readability perspective, seeing the argument `BufferType.Read` at the callsite would be much easier to interpret than `False`.

Reply:
I like this idea, but unfortunately it seems enum doesn't work well with FFI. We can't register a function accepting enum params as packed func

Reply:
Hmm, rats.  Does it work if it's only on the python side, and gets unwrapped into a boolean for the FFI?  That would help the readability in the majority of the cases, but wouldn't require the FFI to be extended to support enums.

Reply:
That should work, but it can make python and c++ api inconsistent. Another way is to use C-style enum, and having FFI to accept `int` for enum values.